applications/matter_weather_station/src/app_event.h

```diff
@@ -6,18 +6,44 @@
 
 #pragma once
 
+#include <cstdint>
+
 struct AppEvent {
-	enum class Type {
-		kButtonPush,
-		kButtonRelease,
-		kTimer,
+	enum FunctionEventType : uint8_t {
+		FunctionPress,
+		FunctionRelease,
+		FunctionTimer,
+		MeasurementsTimer,
+	};
+
+	enum UpdateLedStateEventType : uint8_t { UpdateLedState = MeasurementsTimer + 1 };
+
+	enum OtherEventType : uint8_t {
 #ifdef CONFIG_MCUMGR_SMP_BT
-		kStartSMPAdvertising
+		StartSMPAdvertising = UpdateLedState + 1
 #endif
 	};
 
-	using Handler = void (*)(AppEvent *);
+	AppEvent() = default;
+
+	explicit AppEvent(FunctionEventType type) : Type(type)
+	{
+	}
 
-	Type mType;
-	Handler mHandler;
+	AppEvent(UpdateLedStateEventType type, LEDWidget *ledWidget)
+		: Type(type), UpdateLedStateEvent{ ledWidget }
+	{
+	}
+
+	explicit AppEvent(OtherEventType type) : Type(type)
+	{
+	}
+
+	uint8_t Type;
+
+	union {
```

Comment:
Why we need this annonymous union here?

Reply:
We don't, that was just a copy paste from other samples, where we have more than one argument type passed to the event. Removed where it was unnecessary (weather station and template)